### PR TITLE
Kconfig.tfm: Improvements and fixes

### PR DESCRIPTION
**Pull request by oyvindronningstad**
Make it easier to add TF-M in an app by adding selects so the app
doesn't have to specify the configs.
Remove unneeded configs from the prj.conf files in the tfm samples.

Also, fix case in key file name.

I don't know the reason for a lot of the removed configs, so they may well be needed/wanted (please shout out if they are), I just want to see whether we can simplify things.

### CODE REVIEW

**Review by microbuilder — APPROVED** [review]
Good idea, thanks for the cleanup.

**Comment by microbuilder** [comment]
@oyvindronningstad In addition to the nRF chips, you can always test this with QEMU/MPS2 with some variation of `$ west build -p -b mps2_an521_nonsecure zephyr/samples/tfm_integration/psa_level_1/ -t run`.

File: modules/Kconfig.tfm [hunk 1]
```diff
@@ -5,7 +5,13 @@
...
 menuconfig BUILD_WITH_TFM
 	bool "Build with TF-M as the Secure Execution Environment"
-	depends on TRUSTED_EXECUTION_NONSECURE
+	select CMSIS_RTOS_V2
+	select POLL
+	select THREAD_NAME
+	select THREAD_STACK_INFO
+	select INIT_STACKS
+	select THREAD_MONITOR
+	select TRUSTED_EXECUTION_NONSECURE
```
Comment (ioannisg) [inline modules/Kconfig.tfm:14]: I am not sure this is correct (need to think, at least)
I think the opposite: BUILD_WITH_TFM shall depend on TRUSTED_EXECUTION_NONSECURE.

TRUSTED_EXECUTION_NONSECURE comes directly from the board -ns- definition, and shall switch in all non-secure stuff (including the ability to build with TFM)

Comment (ioannisg) [reply modules/Kconfig.tfm:14]: The user selects to build a Non-Secure Zephyr application by defining an "ns" version of a build target. This currently sets CONFIG_TRUSTED_EXECUTION_NONSECURE=y.  You can build a non-secure Zephyr with or without TFM support. That's why I thin the reverse is more natural dependency order.

**Comment by ioannisg** [comment]
@oyvindronningstad I think this is ready for review (no reason to be draft). Good cleanup, but I would like us to agree on the dependency issue I raised 👍

File: modules/Kconfig.tfm (hunk 1 above)
Comment (oyvindronningstad) [reply modules/Kconfig.tfm:14]: I agree, will fix

File: modules/Kconfig.tfm [hunk 2]
```diff
@@ -5,7 +5,13 @@
 
 menuconfig BUILD_WITH_TFM
 	bool "Build with TF-M as the Secure Execution Environment"
-	depends on TRUSTED_EXECUTION_NONSECURE
+	select CMSIS_RTOS_V2
```
Comment (ioannisg) [inline modules/Kconfig.tfm:8]: @microbuilder @oyvindronningstad I wonder whether these options should be forced here.
I think not, actually. There's no reason for Kernel options like THREAD_STACK_INFO or INIT_STACKS to be selected in the main Kconfig of TFM module. They are sort of un-related.

A compromise would be to switch to "imply", so the user can still override them.

Comment (ioannisg) [reply modules/Kconfig.tfm:8]: Talking about THREAD_STACK_INFO, INIT_STACKS, THREAD_MONITOR, mostly.

Comment (oyvindronningstad) [reply modules/Kconfig.tfm:8]: I'm happy with `imply`. Changed now.

File: modules/Kconfig.tfm (hunk 1 above)
Comment (oyvindronningstad) [reply modules/Kconfig.tfm]: Done

**Comment by oyvindronningstad** [comment]
Tested on nrf, an521 via QEMU, and test built lcpexpresso55s69

File: modules/Kconfig.tfm (hunk 2 above)
Comment (microbuilder) [reply modules/Kconfig.tfm:8]: Agree with `imply`, thanks.

**Comment by ioannisg** [comment]
Just rebased it @oyvindronningstad .

**Review by ioannisg — APPROVED** [review]

**Pull request merged**